File: csharp-sdk/src/Cleanster/Exceptions/AuthException.cs
namespace Cleanster.Exceptions;

/// <summary>
/// Thrown when the API responds with HTTP 401 Unauthorized.
///
/// Indicates an invalid or missing partner access key or user bearer token.
/// Inspect <see cref="ResponseBody"/> for the raw error message from the API.
/// </summary>
public sealed class AuthException : CleansterException
{
    /// <summary>HTTP status code — always 401.</summary>
    public int StatusCode { get; }

    /// <summary>Raw response body returned by the API.</summary>
    public string ResponseBody { get; }

    /// <param name="statusCode">HTTP status code (always 401).</param>
    /// <param name="responseBody">Raw response body from the API.</param>
    /// <param name="message">Human-readable description.</param>
    public AuthException(
        int    statusCode   = 401,
        string responseBody = "",
        string message      = "Authentication failed (HTTP 401).")
        : base(message)
    {
        StatusCode   = statusCode;
        ResponseBody = responseBody;
    }
}

File: php-sdk/src/Exceptions/AuthException.php
<?php

declare(strict_types=1);

namespace Cleanster\Exceptions;

/**
 * Thrown when the API responds with HTTP 401 Unauthorized.
 *
 * This indicates an invalid or missing partner access key or user bearer token.
 * Inspect $responseBody for the raw API error message.
 */
class AuthException extends CleansterException
{
    /**
     * @param int    $statusCode   Always 401.
     * @param string $responseBody Raw response body from the API.
     */
    public function __construct(
        public readonly int    $statusCode,
        public readonly string $responseBody,
        string                 $message = 'Authentication failed (HTTP 401).',
    ) {
        parent::__construct($message, $statusCode);
    }
}
